libs/os/os.go

```diff
@@ -15,13 +16,12 @@ type logger interface {
 
 // TrapSignal catches SIGTERM and SIGINT, executes the cleanup function,
 // and exits with code 0.
-func TrapSignal(logger logger, cb func()) {
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+func TrapSignal(ctx context.Context, logger logger, cb func()) {
+	opctx, _ := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
```

Comment:
I think we still want to ensure a cancellation here, probably:

```suggestion
	opctx, opcancel := signal.NotifyContext(ctx, os.Interrupt, syscall.SIGTERM)
```